src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,111 @@
+package duke;
+
+/**
+ * Main class
+ */
+public class Duke {
+
+    /**
+     * Runs the Duke Program
+     *
+     * @param args No Args needed
+     */
+    public static void main(String[] args) {
+        Ui.printWelcome();
+
+        TaskList list = DataStore.loadData();
+
+        Command actionType = Ui.getCommand();
+        String[] parsedInput = Ui.getInputs();
+
+        while (actionType != Command.BYE) {
+            int indexOfList = -1;
+            Boolean wasAddSuccess = null;
+
+
+            switch (actionType) {
+            case LIST:
+                list.print();
+                break;
+            case MARK:
+                indexOfList = Integer.parseInt(parsedInput[0]);
+                list.markComplete(indexOfList);
+                System.out.println("Marked as complete");
+                list.print(indexOfList);
+                break;
+            case UNMARK:
+                indexOfList = Integer.parseInt(parsedInput[0]);
+                list.markIncomplete(indexOfList);
+                System.out.println("Marked as incomplete");
+                list.print(indexOfList);
+                break;
+            case EVENT:
+                try {
+                    wasAddSuccess = list.add(TaskType.EVENT, parsedInput);
```

Comment:
The process of adding different tasks can be abstracted for better readability since it has also been repeated